src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -472,9 +473,11 @@ DistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId,
 
 	List *shardIntervalsOnNode = ShardIntervalsOnWorkerGroup(workerNode, relationId);
 
+	bool optimizePartitionCalculations = false;
```

Comment:
Why don't we pass true here? This would partially resolve https://github.com/citusdata/citus/issues/4648 as the main reason for the slowness there again is this function -- as far as I can rememer

Reply:
Didn't investigate the reason but tests are failing when passing true here

Reply:
Might be good to at least understand why the tests are failing, in case we have any bug in the code, we might catch with those tests

Reply:
Probably because `SizeQueryType` for this function might be set to `CSTORE_TABLE_SIZE` and we don't have a `worker_partitioned_*_size` function for that type.

Reply:
Correction: The logic for citus size UDFs passes through here and for these functions we shouldn't include the sizes of child tables. (That's how PG size functions work also) However, our new `worker_partitioned_*_size` UDFs include child tables (that's their point actually), so we should pass this `optimizePartitionCalculations` parameter `false` here, since it would trigger our  `worker_partitioned_*_size` functions.